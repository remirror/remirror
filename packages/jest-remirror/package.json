{
  "name": "jest-remirror",
  "version": "3.0.0-beta.6",
  "description": "A text editor for react built with prosemirror",
  "keywords": [
    "prosemirror",
    "react",
    "text editor"
  ],
  "homepage": "https://github.com/remirror/remirror/tree/HEAD/packages/jest-remirror",
  "repository": {
    "type": "git",
    "url": "https://github.com/remirror/remirror.git",
    "directory": "packages/jest-remirror"
  },
  "license": "MIT",
  "contributors": [
    "Ifiok Jr. <ifiokotung@gmail.com>"
  ],
  "type": "module",
  "exports": {
    ".": {
      "types": "./dist-types/index.d.ts",
      "import": "./dist/jest-remirror.js",
      "require": "./dist/jest-remirror.cjs"
    },
    "./environment": {
      "types": "./dist-types/environment.d.ts",
      "import": "./environment/dist/jest-remirror-environment.js",
      "require": "./environment/dist/jest-remirror-environment.cjs"
    },
    "./package.json": "./package.json"
  },
  "main": "./dist/jest-remirror.cjs",
  "module": "./dist/jest-remirror.js",
  "types": "./dist-types/index.d.ts",
  "files": [
    "dist",
    "dist-types",
    "environment"
  ],
  "scripts": {
    "build": "remirror-cli build"
  },
  "dependencies": {
    "@babel/runtime": "^7.22.3",
    "@remirror/core": "3.0.0-beta.6",
    "@remirror/dom": "3.0.0-beta.6",
    "@remirror/pm": "3.0.0-beta.4",
    "@remirror/preset-core": "3.0.0-beta.6",
    "@testing-library/dom": "^9.3.0",
    "@types/sanitize-html": "^2.9.0",
    "jest-prosemirror": "3.0.0-beta.4",
    "sanitize-html": "^2.10.0",
    "tiny-warning": "^1.0.3"
  },
  "devDependencies": {
    "@remirror/cli": "1.1.0-beta.0",
    "@types/node": "^16.18.34",
    "jest": "^29.5.0"
  },
  "peerDependencies": {
    "@types/node": "*",
    "jest": ">= 24.9.0"
  },
  "peerDependenciesMeta": {
    "@types/node": {
      "optional": true
    },
    "jest": {
      "optional": true
    }
  },
  "publishConfig": {
    "access": "public",
    "exports": {
      ".": {
        "types": "./dist/jest-remirror.d.ts",
        "import": "./dist/jest-remirror.js",
        "require": "./dist/jest-remirror.cjs"
      },
      "./environment": {
        "types": "./environment/dist/jest-remirror-environment.d.ts",
        "import": "./environment/dist/jest-remirror-environment.js",
        "require": "./environment/dist/jest-remirror-environment.cjs"
      },
      "./package.json": "./package.json"
    },
    "main": "./dist/jest-remirror.cjs",
    "module": "./dist/jest-remirror.js",
    "type": "module",
    "types": "./dist/jest-remirror.d.ts"
  },
  "@remirror": {
    "tsconfigs": {
      "src": {
        "compilerOptions": {
          "types": [
            "node",
            "jest"
          ]
        }
      }
    }
  },
  "preconstruct": {
    "entrypoints": [
      "index.ts",
      "environment.ts"
    ]
  }
}
