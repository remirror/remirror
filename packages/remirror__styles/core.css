/**
 * Styles extracted from: packages/remirror__theme/src/core-theme.ts
 */
.remirror-editor.ProseMirror {
  word-wrap: break-word;
  white-space: pre-wrap;
  white-space: break-spaces;
  position: relative;
  font-variant-ligatures: none;
  font-feature-settings: 'liga' 0;
  overflow-y: scroll;
}

.remirror-editor.ProseMirror pre {
  white-space: pre-wrap;
}

.remirror-editor.ProseMirror li {
  position: relative;
}

.remirror-editor.ProseMirror[contenteditable='false'] {
  white-space: normal;
}

.remirror-editor.ProseMirror[contenteditable='true'] {
  white-space: pre-wrap;
}

.remirror-editor.ProseMirror hr {
  border-color: #2e2e2e;
}

/* Protect against generic img rules. See also https://github.com/ProseMirror/prosemirror-view/commit/aaa50d592074c8063fc2ef77907ab6d0373822fb */

.remirror-editor.ProseMirror img.ProseMirror-separator {
  display: inline !important;
  border: none !important;
  margin: 0 !important;
}
.remirror-editor.ProseMirror-hideselection *::-moz-selection {
  background: transparent;
  color: inherit;
}
.remirror-editor.ProseMirror-hideselection *::selection {
  background: transparent;
  color: inherit;
}
.remirror-editor.ProseMirror-hideselection *::-moz-selection {
  background: transparent;
  color: inherit;
}
.remirror-editor.ProseMirror-hideselection {
  caret-color: transparent;
}
.remirror-editor .ProseMirror-selectednode {
  outline: 2px solid #8cf;
}
/* Make sure li selections wrap around markers */
.remirror-editor li.ProseMirror-selectednode {
  outline: none;
}
.remirror-editor li.ProseMirror-selectednode:after {
  content: '';
  position: absolute;
  left: -32px;
  right: -2px;
  top: -2px;
  bottom: -2px;
  border: 2px solid #8cf;
  pointer-events: none;
}
