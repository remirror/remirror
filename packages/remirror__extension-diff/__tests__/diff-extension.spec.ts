import { jest } from '@jest/globals';
import { extensionValidityTest, renderEditor } from 'jest-remirror';
import { entries, GetHandler } from '@remirror/core';

import { DiffExtension, DiffOptions } from '../';

extensionValidityTest(DiffExtension);

function create(options?: DiffOptions, handlers: GetHandler<DiffOptions> = {}) {
  const extension = new DiffExtension(options);
  const {
    add,
    nodes: { p, doc },
    view,
    commands,
    helpers,
  } = renderEditor<DiffExtension>([extension]);
  const node = doc(p('This is not yet committed'));

  for (const [name, handler] of entries(handlers)) {
    extension.addHandler(name, handler);
  }

  return { add, p, doc, view, commands, node, helpers };
}

test('#commitChange', () => {
  const { add, commands, node, helpers } = create();

  const { insertText } = add(node);

  commands.commitChange('first commit');
  expect(helpers.getCommits()).toHaveLength(1);

  insertText('\nmore stuff');
  commands.commitChange('commit two');

  expect(helpers.getCommits().map((commit) => commit.message)).toMatchInlineSnapshot(`
    Array [
      "first commit",
      "commit two",
    ]
  `);
});

describe('#revertCommit', () => {
  it('can revert the commit', () => {
    const { add, commands, node, helpers, view } = create();

    const { insertText } = add(node);
    commands.commitChange('first commit');

    insertText('\nmore stuff');
    commands.commitChange('commit two');

    commands.revertCommit();

    expect(helpers.getCommits().map((commit) => commit.message)).toMatchInlineSnapshot(`
      Array [
        "first commit",
        "commit two",
        "Revert: 'commit two'",
      ]
    `);

    expect(view.state.doc).toEqualRemirrorDocument(node);
  });

  it('can revert any of the commits', () => {
    const { add, commands, node, helpers, view, doc, p } = create();

    const { insertText } = add(node);
    commands.commitChange('first commit');

    insertText('\nmore stuff');
    commands.commitChange('commit two');
    insertText(' final changes');
    commands.commitChange('final commit');

    const commit = helpers.getCommit(1);
    commands.revertCommit(commit);

    expect(helpers.getCommits().map((commit) => commit.message)).toMatchInlineSnapshot(`
      Array [
        "first commit",
        "commit two",
        "final commit",
        "Revert: 'commit two'",
      ]
    `);

    expect(view.state.doc).toEqualRemirrorDocument(
      doc(p('This is not yet committed final changes')),
    );
  });
});

describe('#highlightCommit', () => {
  it('can highlight and remove highlights from commits', () => {
    const { add, commands, node, view } = create();

    const { insertText } = add(node);
    commands.commitChange('first commit');
    insertText(' more stuff');
    commands.commitChange('commit two');
    insertText(' final changes');
    commands.commitChange('final commit');

    commands.highlightCommit(1);
    expect(view.dom).toMatchSnapshot();

    commands.highlightCommit('last');
    expect(view.dom).toMatchSnapshot();

    commands.removeHighlightedCommit('last');
    expect(view.dom).toMatchSnapshot();
  });
});

describe('handlers', () => {
  it('can call selection handlers', () => {
    const mocks = { onSelectCommits: jest.fn(), onDeselectCommits: jest.fn() };
    const { add, commands, node } = create(undefined, mocks);
    const { insertText } = add(node);

    commands.commitChange('first commit');

    insertText(' welcome');
    commands.commitChange('commit two');
    expect(mocks.onSelectCommits).toHaveBeenCalled();

    insertText(' not committed').jumpTo('end');

    // TODO this is broken
    expect(mocks.onDeselectCommits).toHaveBeenCalledTimes(1);
  });
});
