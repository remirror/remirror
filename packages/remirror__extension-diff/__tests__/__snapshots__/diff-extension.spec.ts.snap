// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#highlightCommit can highlight and remove highlights from commits 1`] = `
<div
  aria-label=""
  aria-multiline="true"
  class="ProseMirror remirror-editor"
  contenteditable="true"
  role="textbox"
  translate="no"
>
  <p>
    This is not yet committed
    <span
      class="blame-marker"
    >
       more stuff
    </span>
     final changes
  </p>
</div>
`;

exports[`#highlightCommit can highlight and remove highlights from commits 2`] = `
<div
  aria-label=""
  aria-multiline="true"
  class="ProseMirror remirror-editor"
  contenteditable="true"
  role="textbox"
  translate="no"
>
  <p>
    This is not yet committed
    <span
      class="blame-marker"
    >
       more stuff
    </span>
    <span
      class="blame-marker"
    >
       final changes
    </span>
  </p>
</div>
`;

exports[`#highlightCommit can highlight and remove highlights from commits 3`] = `
<div
  aria-label=""
  aria-multiline="true"
  class="ProseMirror remirror-editor"
  contenteditable="true"
  role="textbox"
  translate="no"
>
  <p>
    This is not yet committed
    <span
      class="blame-marker"
    >
       more stuff
    </span>
     final changes
  </p>
</div>
`;
